src/igd_async.rs

```diff
@@ -80,12 +116,19 @@ impl Gateway {
         let gateway = self.inner.clone();
         let (tx, rx) = oneshot::channel();
 
-        let _ = thread::spawn(move || {
+        let jh = thread::spawn(move || {
             let res = gateway.remove_port(protocol, port);
             tx.send(res)
         });
 
-        rx.map_err(|_| panic!("worker thread didn't send the result?"))
+        rx.map_err(|_| {
```

Comment:
Duplicate code - exactly the same semantics as above with the same notes etc.
Maybe we could extract this whole "run in thread and return result as future" into some helper and reuse it?

Reply:
I could add something like that to future-utils I guess.